server/src/main/java/org/apache/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -483,6 +505,28 @@ public boolean isShowServlet()
           return false;
         }
 
+
+        @Override
+        public void generateAcceptableResponse(
+            Request baseRequest,
+            HttpServletRequest request,
+            HttpServletResponse response,
+            int code,
+            String message,
+            String contentType
+        ) throws IOException
+        {
+          if (config.isDefaultJettyErrorResponseInJson()) {
```

Comment:
Regardless of what the new config is set to, we should always use the Accept header to determine the response format right? The new config should only apply when the Accept header is not given.